src/earn/hooks.ts

```diff
@@ -40,3 +48,56 @@ export function useEarnPosition(positionId: string = networkConfig.aaveArbUsdcTo
     return pool
   }, [pools, positionId])
 }
+
+export function useDepositEntrypointInfo({
+  allTokens,
+  pool,
+}: {
+  allTokens: TokenBalances
+  pool: EarnPosition
+}) {
+  const { networkId, dataProps } = pool
+
+  const { swappableFromTokens } = useSwappableTokens()
+  const cashInTokens = useCashInTokens()
+  const isSwapEnabled = useSelector(isAppSwapsEnabledSelector)
+  const userLocation = useSelector(userLocationDataSelector)
+
+  const canDeposit = useMemo(() => {
+    return allTokens[dataProps.depositTokenId]?.balance?.gt(0) ?? false
+  }, [pool, allTokens])
+  const canSameChainSwapToDeposit = useMemo(() => {
+    return (
+      isSwapEnabled &&
+      !!swappableFromTokens.find(
+        (tokenInfo) =>
+          tokenInfo.networkId === networkId && tokenInfo.tokenId !== dataProps.depositTokenId
+      )
+    )
+  }, [pool, isSwapEnabled, swappableFromTokens])
+  const canCrossChainSwap = useMemo(() => {
+    return (
+      getFeatureGate(StatsigFeatureGates.ALLOW_CROSS_CHAIN_SWAPS) &&
+      isSwapEnabled &&
+      !!swappableFromTokens.find((tokenInfo) => tokenInfo.networkId !== networkId)
+    )
+  }, [pool, isSwapEnabled, swappableFromTokens])
+  const canCashIn = useMemo(() => {
+    return !!cashInTokens.find((tokenInfo) => tokenInfo.tokenId === dataProps.depositTokenId)
+  }, [pool, cashInTokens])
+
+  const asyncExchanges = useAsync(async () => {
+    try {
+      const availableExchanges = await fetchExchanges(
+        userLocation.countryCodeAlpha2,
+        dataProps.depositTokenId
+      )
+
+      return availableExchanges
+    } catch (error) {
+      return []
+    }
+  }, [])
+  const exchanges = asyncExchanges.result ?? []
```

Comment:
Will this work? Does it need to be in a useMemo so that it changes to the results once they are available?